### ISSUE TITLE
Impl `IntoIterator` for `&mut Vector<T>`

### ISSUE THREAD

**Issue by Carnagion** [jneem/imbl#79]
I noticed that `&mut Vector<T>` does not implement `IntoIterator<Item = &mut T>` - however, `&Vector<T>` impls `IntoIterator<Item = &T>`.

`Vector<T>`'s API is thus slightly inconsistent with itself and with `std::vec::Vec<T>`'s API, since the latter impls `IntoIterator<Item = &mut T>` for its mutably borrowed version.

Furthermore, it is definitely possible for this impl to exist, because `Vector::<T>::iter_mut` exists. An `IntoIterator` impl for `&mut Vector<T>` thus just needs to have `type IntoIter = IterMut<'_, T>`, much like in the `IntoIterator` impl for `&Vector<T>`.

This should be a simple fix of a couple lines:
```rs
impl<'a, T> IntoIterator for &'a mut Vector<T> {
    type Item = &'a mut T;

    type IntoIter = IterMut<'a, T>;

    fn into_iter(self) -> Self::IntoIter {
        self.iter_mut()
    }
}
```

I am willing to make a PR for this if necessary.

**Comment by jneem** [comment]
Sure, this makes sense to me. I'd be happy to merge a PR. I can probably also manage it myself, but not until next week...

**Comment by Carnagion** [comment]
Great, I've opened a PR (#80).

**Comment by jneem** [comment]
Closed by #80
